### PR TITLE
ticdc: add index design for cdc storage sink

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by nongfushanquan — APPROVED** [review]

**Review by zhaoxinyu — APPROVED** [review]

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 1]
```diff
@@ -101,6 +97,22 @@ URI 中其他可配置的参数如下：
 > - 发生 DDL 操作后，表的版本为该 DDL 在上游 TiDB 执行结束的 TSO。但是，表版本的变化并不意味着表结构的变化。例如，在表中的某一列添加注释，不会导致 `schema.json` 文件内容发生变化。
 > - 进程重启，表的版本为进程重启时 Changefeed 的 checkpoint TSO。在有很多表的情况下，重启时需要遍历所有目录并找到上一次重启时每张表写入的位置，这样的操作耗时较长。因此，TiCDC 选择在一个以 checkpoint TSO 为版本的新目录下写入数据，而不是在旧版本的目录下继续写入数据。
 
+### Index 文件
+
```
Comment (qiancai) [inline ticdc/ticdc-sink-to-cloud-storage.md:101]: 这里是否需要说明下什么是 Index 文件

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 2]
```diff
@@ -101,6 +97,22 @@ URI 中其他可配置的参数如下：
...
+Index 文件与数据变更记录存储在相同路径：
```
Comment (CharlesCheung96) [inline ticdc/ticdc-sink-to-cloud-storage.md:102]:
```suggestion
Index 文件用于防止已写入的数据被错误覆盖，与数据变更记录存储在相同路径：
```

**Review by qiancai — APPROVED** [review]

**Review by Frank945946 — APPROVED** [review]

File: ticdc/ticdc-sink-to-cloud-storage.md [hunk 3]
```diff
@@ -101,6 +97,22 @@ URI 中其他可配置的参数如下：
...
+{scheme}://{prefix}/{schema}/{table}/{table-version-separator}/{partition-separator}/{date-separator}/CDC.index
+```
+
+Index 文件记录了当前目录下所使用到的最大文件名，比如：
+
+```
+CDC000005.csv
+```
+
+上述内容表明该目录下 CDC000001.csv 到 CDC000004.csv 文件已被占用，当 TiCDC 集群中发生表调度或者节点重启时，新的节点会读取 Index 文件，并判断 CDC000005.csv 是否被占用。如果未被占用，则新节点会从 CDC000005.csv 开始写文件。如果已被占用，则从 CDC000006.csv 开始写文件，这样可防止覆盖其他节点写入的数据。
```
Comment (hfxsd) [inline ticdc/ticdc-sink-to-cloud-storage.md:114]:
```suggestion
上述内容表明该目录下 `CDC000001.csv` 到 `CDC000004.csv` 文件已被占用，当 TiCDC 集群中发生表调度或者节点重启时，新的节点会读取 Index 文件，并判断 `CDC000005.csv` 是否被占用。如果未被占用，则新节点会从 `CDC000005.csv` 开始写文件。如果已被占用，则从 `CDC000006.csv` 开始写文件，这样可防止覆盖其他节点写入的数据。
```

**Pull request merged**